src/util/invariant.h

```diff
@@ -225,15 +225,26 @@ inline void invariant_violated_string(
 // for INVARIANT.
 
 // The condition should only contain (unmodified) arguments to the method.
+// Inputs include arguments passed to the function, as well as member
+// variables that the function may read.
 // "The design of the system means that the arguments to this method
 //  will always meet this condition".
+//
+// It is also supposed to signal that even though an object is in a
+// valid state, a particular function may not be operating on objects
+// with the particular properties this one has.
```

Comment:
"this particular function assumes that the inputs have the given properties" ?

The point is that PRECONDITION documents / checks assumptions about the inputs that it is the caller's responsibility to make happen before the call.

Reply:
I like your wording better, I will change it. Thanks.